src/tekton/tektonitem.ts

```diff
@@ -126,63 +131,136 @@ export abstract class TektonItem {
     return workspace.getConfiguration('vs-tekton')['vs-tekton.outputFormat'];
   }
 
-  static async startObject(context: Trigger[], message: string): Promise<StartObject> {
-    const resources: QuickPickItem[] | undefined = context[0].resources ? context[0].resources.map<QuickPickItem>(label => ({ label: label.name, resourceType: label['resourceType'] ? label['resourceType'] : undefined })) : undefined;
-    const params: QuickPickItem[] | undefined = context[0].params ? context[0].params.map<QuickPickItem>(label => ({ label: label.name })) : undefined;
-    const workspaces: QuickPickItem[] | undefined = context[0].workspaces ? context[0].workspaces.map<QuickPickItem>(label => ({ label: label.name })) : undefined;
-
-    const title = `Start ${message}`;
-
-    interface Ref {
-      name: string;
-      type: string;
+  static async PipelineResourceReturn(name: string, context: Trigger[]): Promise<Ref[]> {
+    let pipeR: TknPipelineResource[] = [];
+    const element = context[0].resources.find(e => e.name === name);
+    const kubectl = await k8s.extension.kubectl.v1;
```

Comment:
Now k8s extension are optional, you cannot relay on it

Reply:
okay, I will remove k8s dependency from here.